tests/testthat/test-generate_data.R

```diff
@@ -280,3 +280,38 @@ test_that("genFactor works.", {
   expect_length(genFactor(copy(dt), c("q1", "q2"), replace = TRUE), 4)
   expect_equal(genFactor(copy(dt), c("q1", "q2"), labels = labels), dt_res)
 })
+
+#genDummy ----
+test_that("genDummy throws errors.", {
+  oldSeed <- .Random.seed
+  set.seed(10076)
+  
+  d1 <- defData(varname = "rx", formula = "1;1", dist = "trtAssign")
+  d1 <- defData(d1, varname = "male", formula = .4, dist = "binary")
+  d1 <- defData(d1, varname = "z", formula = "0 - 1.2*rx - 1*male", dist = "nonrandom")
+  
+  dd <- genData(5, d1)
+  
+  # Initial data checks
+  expect_error(genDummy(varname = "rx", sep = ".", replace = FALSE))
```

Comment:
[`expect_error`](https://testthat.r-lib.org/reference/expect_error.html#arguments) has some features that allow you to test for specific errors, either by regex matching against the error message or by checking if the error is of the correct class e.g. [simstudy::missingArgument](https://github.com/kgoldfeld/simstudy/blob/main/R/conditions.R#L29), this is important as it prevents the test from passing in case there is some other error you were not expecting.